source/firstprogram-expanding-network.rst

```diff
@@ -181,159 +185,324 @@ With these values defined, ``create24HourSchedules`` can be made. However, when
 
 .. note:: The timings of departure and return are to be made in the 24 hour format.  
 
-The whole main file code is 
+
+Handling Current Locations
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Now, we also need to handle the location of the individual at every step to ensure that the individual is only in contact with the people in the same location. This is done by adding a new ``currentLocation`` parameter in the person class. We modify the constructor to include this parameter.
 
 .. code-block:: scala 
 
-  package sir
-
-  import java.util.Date
-  import com.bharatsim.engine.ContextBuilder._
-  import com.bharatsim.engine._
-  import com.bharatsim.engine.actions.StopSimulation
-  import com.bharatsim.engine.basicConversions.decoders.DefaultDecoders._
-  import com.bharatsim.engine.basicConversions.encoders.DefaultEncoders._
-  import com.bharatsim.engine.dsl.SyntaxHelpers._
-  import com.bharatsim.engine.execution.Simulation
-  import com.bharatsim.engine.graph.ingestion.{GraphData, Relation}
-  import com.bharatsim.engine.graph.patternMatcher.MatchCondition._
-  import com.bharatsim.engine.listeners.{CsvOutputGenerator, SimulationListenerRegistry}
-  import com.bharatsim.engine.models.Agent
-  import com.bharatsim.engine.utils.Probability.biasedCoinToss
-  import com.bharatsim.examples.epidemiology.sir.InfectionStatus._
-  import com.typesafe.scalalogging.LazyLogging
-
-  object Main extends LazyLogging {
-    private val initialInfectedFraction = 0.01
+    case class Person(id: Long, age: Int, infectionState: InfectionStatus, infectionDay: Int, currentLocation: String = "") extends Agent {
 
-    private val myTick: ScheduleUnit = new ScheduleUnit(1)
-    private val myDay: ScheduleUnit = new ScheduleUnit(myTick * 24)
+Then, we add a behaviour to update the current location of the person at every step.
 
-    def main(args: Array[String]): Unit = {
-
-      var beforeCount = 0
-      val simulation = Simulation()
+.. code-block:: scala 
 
-      simulation.ingestData(implicit context => {
-        ingestCSVData("citizen10k.csv", csvDataExtractor)
-        logger.debug("Ingestion done")
-      })
+    private def getCurrentLocation(context: Context): Option[String] = {
+      val schedule = context.fetchScheduleFor(this).get
+      val currentStep = context.getCurrentStep
+      val placeType: String = schedule.getForStep(currentStep)
 
-      simulation.defineSimulation(implicit context => {
-        create24HourSchedules()
+      Some(placeType)

```

Comment:
Could you add one line below this explaining what ``Option`` and ``Some`` mean? Nothing too elaborate, just a small description like from the link you sent me (https://stackoverflow.com/questions/49944872/understanding-some-and-option-in-scala). Add a link to some online documentation on this as well, if possible.